lib/internal/modules/cjs/loader.js

```diff
@@ -611,9 +618,14 @@ Module._findPath = function(request, paths, isMain) {
 
   // For each path
   for (let i = 0; i < paths.length; i++) {
-    // Don't search further if path doesn't exist and request is inside the path
+    // Don't search further if path doesn't exist
+    // or doesn't have permission to it
     const curPath = paths[i];
-    if (insidePath && curPath && _stat(curPath) < 1) continue;
+    if (insidePath && curPath &&
+      ((permission.isEnabled() && permission.has('fs.read', curPath) === false) || _stat(curPath) < 1)
```

Comment:
may be, `permission.isEnabled()` can be performed once before the loop entry, and the result cached?

Reply:
I'll do it. But I think that won't change much, because `isEnabled()` will be optimized to a cached result I assume. Once it's true, it will always be true.

Reply:
Applied ce3c5d267a99aa1c6c0143976f025460a818f1c8